syntax/parse.rs

```diff
@@ -567,7 +567,21 @@ fn parse_extern_fn(
                     });
                     continue;
                 }
-                return Err(Error::new_spanned(arg, "unsupported signature"));
+
+                match lang {
+                    Lang::Rust => {
```

Comment:
This is branching on the wrong thing -- the requirement for Pin i.e. the use of `&mut self` vs `self: Pin<&mut Self>` is not based on the source language of the function implementation, but on whether the receiver type is or is not opaque to Rust. Consider for example:

```rust
use core::pin::Pin;

#[cxx::bridge]
pub mod ffi {
    extern "C++" {
        type C;
    }

    extern "Rust" {
        type R;
        fn f(self: &mut R);
        fn f(self: Pin<&mut C>);  // Rust method on an opaque C++ type
    }
}

pub struct R;

impl R {
    fn f(&mut self) {}
}

impl ffi::C {
    fn f(self: Pin<&mut Self>) {}
}
```

where the C struct is:

```cpp
class C {
  void f() noexcept;
};
```